File: core/src/main/java/reader/ecos/org/core/io/files/ZipNavigatorServiceImpl.java
package reader.ecos.org.core.io.files;

import reader.ecos.org.core.io.files.exceptions.DoNoExistsExcepiton;

@SuppressWarnings("unused")
public class ZipNavigatorServiceImpl implements ZipNavigatorService {
    @Override
    public FileCollection getFilesInside(ZipFile file, Path path) throws DoNoExistsExcepiton {

        if(!file.exists())
            throw new DoNoExistsExcepiton("");

        return new FileCollectionImpl();
    }

    @Override
    public DirectoryCollection getDirectoriesInside(ZipFile file, Path path) {
        return new DirectoryCollectionImpl();
    }
}

File: files/src/main/kotlin/org/ecos/core/files/ZipNavigatorServiceImpl.kt
package org.ecos.core.files

import org.ecos.reader.core.exceptions.DoNoExistsException

import org.ecos.reader.core.exceptions.UnknownError

@Suppress("unused")
class ZipNavigatorServiceImpl : ZipNavigatorService {
    @Throws(DoNoExistsException::class, UnknownError::class)
    override fun getFilesInside(file: ZipFile): FileCollection {
        if (!file.exists())
            throw DoNoExistsException("")

        return file.files
    }
}